external/mqtt/src/test/scala/org/apache/spark/streaming/mqtt/MQTTStreamSuite.scala

```diff
@@ -121,8 +130,18 @@ class MQTTStreamSuite extends FunSuite with Eventually with BeforeAndAfter {
         val message: MqttMessage = new MqttMessage(data.getBytes("utf-8"))
         message.setQos(1)
         message.setRetained(true)
+
+        def tryPublish(times: Int): Unit =
+          if (times > 0) try {
+            msgTopic.publish(message)
+          } catch {
+            case e: MqttException if e.getReasonCode == MqttException.REASON_CODE_MAX_INFLIGHT =>
+              Thread.sleep(50) // wait for Spark streaming to consume something from the message queue
+              tryPublish(times - 1)
+          }
+
         for (i <- 0 to 100) {
-          msgTopic.publish(message)
+          tryPublish(3)
```

Comment:
I can do that, but it won't retry the same message, instead if would just ignore a failure. In that case the total number of messages will be less than 101.